doc/docs.md

```diff
@@ -4764,6 +4764,21 @@ fn legacy_function() {}
 fn inlined_function() {
 }
 
+// This function's calls will NOT be inlined.
+[noinline]
+fn function() {
+}
+
+// This function will NOT return to its callers.
+// Such functions can be used at the end of or blocks,
+// just like exit/1 or panic/1. Such functions can not
+// have return types, and should end either in for{}, or
+// by calling other `[noreturn]` functions.
+[noreturn]
```

Comment:
In typescript there is a type called `never´. Wouldn't it be good to to have 
```
fn forever() never {}
```
?

It might be more work but it could also have better compiler support.

Reply:
V already has attribute support.  No need to add a new type.

Reply:
> Could you add something that `[noreturn]` is required if functions ends with a [noreturn] function?

It is not required. You can have normal functions that call `[noreturn]` ones. The compiler just can not enforce the restrictions for such functions, so you can not use them in `or {}` blocks.